src/openvic-simulation/scripts/Condition.hpp

```diff
@@ -0,0 +1,233 @@
+#pragma once
+
+#include "openvic-simulation/types/IdentifierRegistry.hpp"
+#include <variant>
+
+namespace OpenVic {
+	struct ConditionManager;
+	struct GameManager;
+
+	#define BITFIELD_OP(type, int_type) \
+		constexpr type operator&(type lhs, type rhs) { \
+			return static_cast<type>(static_cast<int_type>(lhs) & static_cast<int_type>(rhs)); \
+		} \
+		constexpr type operator|(type lhs, type rhs) { \
+			return static_cast<type>(static_cast<int_type>(lhs) | static_cast<int_type>(rhs)); \
+		} \
+		constexpr bool operator==(type lhs, uint8_t rhs) { \
+			return static_cast<type>(static_cast<int_type>(lhs)) == rhs; \
+		} \
+		constexpr bool operator>(type lhs, uint8_t rhs) { \
+			return static_cast<type>(static_cast<int_type>(lhs)) > rhs; \
+		} \
+		constexpr bool operator<(type lhs, uint8_t rhs) { \
+			return static_cast<type>(static_cast<int_type>(lhs)) < rhs; \
+		}
+
+	enum class value_type_t : uint8_t {
+		IDENTIFIER  = 1 << 0,
+		STRING      = 1 << 1,
+		BOOLEAN     = 1 << 2,
+		BOOLEAN_INT = 1 << 3,
+		INTEGER     = 1 << 4,
+		REAL        = 1 << 5,
+		COMPLEX     = 1 << 6,
+		GROUP       = 1 << 7
+	};
+	
+	static constexpr std::string get_value_type_string(value_type_t value) {
+		switch (value) {
+			case value_type_t::IDENTIFIER: return "IDENTIFIER";
+			case value_type_t::STRING: return "STRING";
+			case value_type_t::BOOLEAN: return "BOOLEAN";
+			case value_type_t::BOOLEAN_INT: return "BOOLEAN_INT";
+			case value_type_t::INTEGER: return "INTEGER";
+			case value_type_t::REAL: return "REAL";
+			case value_type_t::COMPLEX: return "COMPLEX";
+			case value_type_t::GROUP: return "GROUP";
+		}
```

Comment:
`get_value_type_string`, `get_scope_string` and `get_identifier_type_string` currently return nothing for inputs that are combinations of multiple bits, and `get_value_type_string` also returns nothing for `0`. If they get any of these inputs the behaviour is undefined and it could crash the program. You can avoid this worst case scenario by adding a `default` case, either returning an empty string or some sort of warning/error message.

Also all 3 of these functions can return `std::string_view`s, raw string literals are stored in the program's read-only memory they won't get freed when they go out of scope. Currently that memory is getting copied into a `std::string` every time the functions are called, which would be avoided by using `std::string_view`.